src/requirements/requirement-graph-builder-from-user-data.ts

```diff
@@ -53,13 +53,13 @@ export default function buildRequirementFulfillmentGraphFromUserData(
           [
             uniqueId,
             {
-              optIn: Array.from(
-                new Set([
+              optIn: [
+                ...new Set([
                   ...Object.keys(choice.arbitraryOptIn),
                   ...choice.acknowledgedCheckerWarningOptIn,
-                ])
-              ),
-              optOut: choice.optOut,
+                ]),
+              ],
+              optOut: [...new Set(choice.optOut)],
```

Comment:
Maintain the invariant that there are no duplicate keys (although there shouldn't be, it doesn't hurt to have built-in redundancy! It shouldn't hurt performance since these lists are almost always on the order of ~0-5 elements long).